src/SimulationMap.cu

```diff
@@ -234,10 +247,9 @@ __device__ int SimulationMap::get_neighbor_node_id(int current_node_id, SpacePoi
     else if(current_face == next_face || !does_face_have_nodes[find_face_index(next_face)])
     {
         // Neighbor node does not exist, but it can be created
-        nodes = device_realloc(nodes, n_of_nodes, n_of_nodes + 1);
+        nodes[n_of_nodes].detach_particle();
         nodes[n_of_nodes] = MapNode(polyhedron, next_face, neighbor_coordinates);
 
-        *nodes_directions = device_realloc(*nodes_directions, n_of_nodes, n_of_nodes + 1);
```

Comment:
Because you don't need to update `*nodes_directions` anymore, you should accept `SpacePoint *nodes_directions` instead of `SpacePoint **nodes_directions` as an argument to this function